### PR TITLE
Bug: fix rate and allocation validation

### PR DESCRIPTION
**Pull request by miikahyttinen**
Fixes:

- Restricted _allocation_ and _rate_ fields' inputs to only numbers
- _Allocation_ can have a value between 0 and 100
- _Rate_ can have value of 0 and greater
- Add error message for invalid _rate_

### CODE REVIEW

**Review by LeadingMoominExpert — APPROVED** [review]

File: components/form/TimesheetForm.tsx [hunk 1]
```diff
@@ -168,37 +171,55 @@ function TimesheetForm({
...
+                                    ...timesheetFields,
+                                    rate: event.target.value
+                                        ? Number(event.target.value)
+                                        : undefined,
+                                })
+                            }}
+                            data-testid="form-field-rate"
+                        />
+                        <FormErrorMessage>
+                            Rate must be a positive number.
```
Comment (mapehe) [inline components/form/TimesheetForm.tsx:216]: Nit: Since zero is allowed, `Rate must be a non-negative number.`

**Review by mapehe — COMMENTED** [review]
LGTM, one super nit comment.

**Comment by mapehe** [comment]
Nice! 💎 

Please rebase this to a single commit. Feel free to merge after that 🙏

File: components/form/TimesheetForm.tsx (hunk 1 above)
Comment (miikahyttinen) [reply components/form/TimesheetForm.tsx]: Changed this 👍

**Review by mapehe — APPROVED** [review]

**Pull request merged**